### PR TITLE
Add Plone 5.1 Support

### LINKED ISSUE #48: Plone 5.1 support
Implement compatibility with plone 5.1
This include ftw.publisher.sender and ftw.publisher.receiver

The support of following dependencies can be dropped for plone 5.1:
- 'collective.geo.contentlocations',
- 'collective.geo.geographer',
- 'ftw.contentpage',
- 'ftw.shop',
- 'simplelayout.base',

### PR DESCRIPTION
**Pull request by Nachtalb**
🚧 UNDER CONSTRUCTION 🚧 
resolves #48 

~~Plone 4.3 continues on own branch, recommendation by @maethu~~

### CODE REVIEW

**Comment by maethu** [comment]
@Nachtalb That was no exactly what I said. We split in two branches only if it is necessary. The preferred way is to support both at the same time. 

We can split it up, but we need good reason for that.

**Comment by Nachtalb** [comment]
@maethu Oh sorry, I understood you wrong. I will change it.

File: ftw/publisher/core/tests/test_geo_data_adapter.py [hunk 1]
```diff
@@ -1,13 +1,19 @@
...
 from zope.annotation import IAttributeAnnotatable
 from zope.component import getAdapter
 from zope.component import queryAdapter
 
+if not IS_PLONE_5:
+    from collective.geo.contentlocations.interfaces import IGeoManager
+    from collective.geo.geographer.interfaces import IGeoreferenceable
+
 
+@skipIf(IS_PLONE_5, 'ftw.shop is not available for plone 5')
```
Comment (jone) [inline ftw/publisher/core/tests/test_geo_data_adapter.py]: Probably `colelctive.geo`, not `ftw.shop`?

File: setup.py [hunk 2]
```diff
@@ -7,36 +7,37 @@
 extras_require = {
     'dexterity': [
         'plone.app.dexterity',
-        ]}
+        ],
+    'plone4': [
```
Comment (jone) [inline setup.py]: Can we change the extras name to `tests_plone4`?
- extras makes only sense when testing ftw.publisher.core itself with Plone 4
- matches the same extras in ftw.publisher.receiver

File: test-plone-5.1.x.cfg [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+package-name = ftw.publisher.core
+
+auto-checkout =
+    plonetheme.onegov
+    ftw.simplelayout
+
+
+[branches]
+plonetheme.onegov = plone5
+ftw.simplelayout = plone-5.1.x
```
Comment (jone) [inline test-plone-5.1.x.cfg]: Are we going to wait for the simplelayout Plone 5 support to be merged?

Comment (Nachtalb) [reply test-plone-5.1.x.cfg]: It needs the new plone 5 simplelayout for the tests to run.
Otheriwse it gets:

Comment (mbaechtold) [reply test-plone-5.1.x.cfg]: I just asked @maethu about this and the problem is, that `ftw.simplelayout` has a dependency on `plonetheme.blueberry` which in turn depends on `ftw.publisher.core` (egg/hen problem). 

We need to leave this as is.

File: ftw/publisher/core/tests/test_geo_data_adapter.py (hunk 1 above)
Comment (mbaechtold) [reply ftw/publisher/core/tests/test_geo_data_adapter.py]: @Nachtalb has changed that.

File: setup.py (hunk 2 above)
Comment (mbaechtold) [reply setup.py]: @Nachtalb has changed that.

**Comment by Nachtalb** [comment]
@mbaechtold rebased onto master

File: docs/HISTORY.txt [hunk 4]
```diff
@@ -12,7 +12,7 @@ Changelog
 -------------------
 
 - Handle broken relations properly. [jone, mbaechtold]
-
+- Add support for Plone 5.1. [Nachtalb]
```
Comment (mbaechtold) [inline docs/HISTORY.txt]: Please move this up to `2.10.3`.

File: test-plone-5.1.x.cfg (hunk 3 above)
Comment (mbaechtold) [reply test-plone-5.1.x.cfg]: @jone, should we merge this?

Comment (jone) [reply test-plone-5.1.x.cfg]: @mbaechtold yes, please merge it 👍

**Pull request merged**